src/github/client/app-token-holder.ts

```diff
@@ -5,13 +5,23 @@ import { AuthToken, ONE_MINUTE, TEN_MINUTES } from "./auth-token";
 import * as PrivateKey from "probot/lib/private-key";
 import LRUCache from "lru-cache";
 import { InstallationId } from "./installation-id";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+import { envVars } from "config/env";
+import { getLogger } from "config/logger";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
+import { createHashWithSharedSecret } from "utils/encryption";
 
 export type KeyLocator = (installationId: InstallationId) => string;
 
 /**
  * By default, we just look for a key in the `PRIVATE_KEY` env var.
  */
+let shouldUseNewSecret = false;
 export const cloudKeyLocator: KeyLocator = () => {
+	booleanFlag(BooleanFlags.USE_NEW_GITHUB_PRIVATE_KEY, false)?.then(newValue => shouldUseNewSecret = newValue);
+	if (shouldUseNewSecret) {
+		return Buffer.from(envVars.PRIVATE_KEY_VAULT, 'base64').toString();
```

Comment:
We are mimicking probot here: https://github.com/probot/get-private-key/blob/094fe45cf040c345802bc81a12dac576b92a4589/src/index.ts#L32

Reply:
Tbh I don't think we should say "we are mimicking probot", as this is required because we put base64 encoded key in vault, not because we are or not using probot.